File: 11. Container With Most Water/solution.cs
public class Solution {
    public int MaxArea(int[] height) {
        int result = 0;

        int leftBorder = 0;
        int rightBorder = height.Length - 1;

        int curArea = 0;
        while (rightBorder > leftBorder)
        {
            int width = rightBorder - leftBorder;
            if (height[leftBorder] > height[rightBorder])
            {
                curArea = height[rightBorder] * width;
                rightBorder--;
            }
            else
            {
                curArea = height[leftBorder] * width;
                leftBorder++;
            }
            if (curArea > result) result = curArea;
        }

        return result;
    }
}

File: 11. Container With Most Water/solution.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int result = 0;

        int l_border = 0;
        int r_border = height.size()-1;

        int curArea;
        while (r_border > l_border) {
            if (height[l_border] > height[r_border]) {
                curArea = height[r_border] * (r_border - l_border); 
                r_border--;
            } else {
                curArea = height[l_border] * (r_border - l_border);
                l_border++;
            }
            if (curArea > result) result = curArea;
        }

        return result;
    }
};